src/Infrastructure.Dapper/Tools/Repositories/SendRepository.cs

```diff
@@ -48,4 +50,56 @@ public async Task<ICollection<Send>> GetManyByDeletionDateAsync(DateTime deletio
             return results.ToList();
         }
     }
+
+    public UpdateEncryptedDataForKeyRotation UpdateForKeyRotation(Guid userId, IEnumerable<Send> sends)
+    {
+        return async (connection, transaction) =>
+        {
+            // Create temp table
+            var sqlCreateTemp = @"
+                            SELECT TOP 0 *
+                            INTO #TempSend
+                            FROM [dbo].[Send]";
+
+            await using (var cmd = new SqlCommand(sqlCreateTemp, connection, transaction))
+            {
+                cmd.ExecuteNonQuery();
+            }
+
+            // Bulk copy data into temp table
+            using (var bulkCopy = new SqlBulkCopy(connection, SqlBulkCopyOptions.KeepIdentity, transaction))
+            {
+                bulkCopy.DestinationTableName = "#TempSend";
+                var sendsTable = sends.ToDataTable();
+                foreach (DataColumn col in sendsTable.Columns)
+                {
+                    bulkCopy.ColumnMappings.Add(col.ColumnName, col.ColumnName);
+                }
+
+                sendsTable.PrimaryKey = new DataColumn[] { sendsTable.Columns[0] };
+                await bulkCopy.WriteToServerAsync(sendsTable);
+            }
```

Comment:
👍🏻 Been a long time since I've seen plain old ADO.Net in use, and this brings me back. I used to have this whole toolkit for splicing together strong types with the type erased world of ADO, and it had lots of code that looked like this, including the lambda-returning business for being able to splice together various things. It also included fancy bits like attributes to signal primary keys, an in-built dynamic pivot engine, and so on. I had a lot of time on my hands back then.

Super happy I picked up this review. 😄

Reply:
Sounds like quite the package! Most of this is pretty old code that I'm keeping around because it works for our use case (if it ain't broke...)

Reply:
> if it ain't broke... 

Then you probably aren't looking deep enough. ;)